java/vector/src/main/java/org/apache/arrow/vector/complex/FixedSizeListVector.java

```diff
@@ -579,10 +579,73 @@ public void transfer() {
 
     @Override
     public void splitAndTransfer(int startIndex, int length) {
+      Preconditions.checkArgument(startIndex + length <= valueCount);
+      final int startPoint = listSize * startIndex;
+      final int sliceLength = listSize * (startIndex + length) - startPoint;
```

Comment:
this can be simplified to listSize * length?

Reply:
Thanks, you are right, fixed.